pkg/p2p/discover.go

```diff
@@ -14,8 +14,15 @@ import (
 	madns "github.com/multiformats/go-multiaddr-dns"
 )
 
+func isDNSProtocol(protoString string) bool {
```

Comment:
I'd use `Code` instead of `Name` and match it against already defined constants as: `multiaddr.P_DNS4`, etc...